### PR TITLE
feat(rpc): Ability to filter HPMNs in masternodelist and protx list rpcs

### PR DESCRIPTION
**Pull request by ogabrielides**
## Issue being fixed or feature implemented
Added the filter `hpmn` for both `masternodelist` and `protx list` rpcs.

## How Has This Been Tested?

Calling this RPC on Testnet.

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: src/rpc/evo.cpp [hunk 1]
```diff
@@ -1364,7 +1365,9 @@ static UniValue protx_list(const JSONRPCRequest& request)
 
         CDeterministicMNList mnList = deterministicMNManager->GetListForBlock(::ChainActive()[height]);
         bool onlyValid = type == "valid";
+        bool onlyHPMN = type == "hpmn";
         mnList.ForEachMN(onlyValid, [&](const auto& dmn) {
+            if (onlyHPMN && dmn.nType == MnType::Regular) return;
```
Comment (UdjinM6) [inline src/rpc/evo.cpp:1370]:
```suggestion
            if (onlyHPMN && dmn.nType != MnType::HighPerformance) return;
```

File: src/rpc/masternode.cpp [hunk 2]
```diff
@@ -609,13 +610,17 @@ static UniValue masternodelist(const JSONRPCRequest& request)
...
     bool showRecentMnsOnly = strMode == "recent";
+    bool showHPMNsOnly = strMode == "hpmn";
     int tipHeight = WITH_LOCK(cs_main, return ::ChainActive().Tip()->nHeight);
     mnList.ForEachMN(false, [&](auto& dmn) {
         if (showRecentMnsOnly && mnList.IsMNPoSeBanned(dmn)) {
             if (tipHeight - dmn.pdmnState->GetBannedHeight() > Params().GetConsensus().nSuperblockCycle) {
                 return;
             }
         }
+        if (showHPMNsOnly && dmn.nType == MnType::Regular) {
```
Comment (UdjinM6) [inline src/rpc/masternode.cpp:621]:
```suggestion
        if (showHPMNsOnly && dmn.nType != MnType::HighPerformance) {
```

**Review by UdjinM6 — APPROVED** [review]
lightly tested ACK (works as expected on testnet)

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Pull request merged**